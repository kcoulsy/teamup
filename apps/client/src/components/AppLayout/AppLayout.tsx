import React, { PropsWithChildren } from 'react';
import { Layout } from 'antd';
import SideNav from '../SideNav/SideNav';

const { Content } = Layout;

function AppLayout({ children }: PropsWithChildren<{}>) {
  return (
    <Layout className='app-layout'>
      <SideNav />
      <Layout>
        <Content className='app-layout__content'>
          <div className='app-layout__container'>{children}</div>
        </Content>
      </Layout>
    </Layout>
  );
}

export default AppLayout;
