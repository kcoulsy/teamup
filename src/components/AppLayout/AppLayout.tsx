import React from 'react';
import { Layout } from 'antd';
import SideNav from './../SideNav/SideNav';

const { Header, Content } = Layout;
export const AppLayout: React.FunctionComponent = ({ children }) => {
    return (
        <Layout style={{ minHeight: '100vh' }}>
            <SideNav />
            <Layout className="site-layout">
                <Header
                    className="site-layout-background"
                    style={{ padding: 0 }}
                />
                <Content style={{ margin: '0 16px' }}>
                    <div
                        className="site-layout-background"
                        style={{ padding: 24, minHeight: 360 }}>
                        {children}
                    </div>
                </Content>
            </Layout>
        </Layout>
    );
};

export default AppLayout;
