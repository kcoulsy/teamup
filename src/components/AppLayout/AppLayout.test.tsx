import React from 'react';
import { shallow } from 'enzyme';

import AppLayout from './AppLayout';
import { shallowToJson } from 'enzyme-to-json';

describe('render AppLayout component', () => {
    it('should render AppLayout component', () => {
        const child = () => <p>JSX Child</p>;
        const wrapper = shallow(<AppLayout>{child}</AppLayout>);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
})