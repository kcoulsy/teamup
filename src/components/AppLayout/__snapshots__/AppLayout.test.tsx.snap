// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AppLayout component should render AppLayout component 1`] = `
<Layout
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <Connect(Component) />
  <Layout
    className="site-layout"
  >
    <Header
      className="site-layout-background"
      style={
        Object {
          "padding": 0,
        }
      }
    />
    <Content
      style={
        Object {
          "margin": "0 16px",
        }
      }
    >
      <div
        className="site-layout-background"
        style={
          Object {
            "minHeight": 360,
            "padding": 24,
          }
        }
      >
        <Component />
      </div>
    </Content>
  </Layout>
</Layout>
`;
